pulpcore/cli/common.py

```diff
@@ -47,25 +71,24 @@ def wait_for_task(self, task_href, timeout=120):
 
 @click.group()
 @click.version_option(prog_name="pulp3 command line interface")
-@click.option("--base-url", default="https://localhost", help="Api base url")
+@click.option("--config", help="Path to the toml config file")
+@click.option("--base-url", help="Api base url")
 @click.option("--user", default="admin", help="Username on pulp server")
 # @click.option("--password", prompt=True, hide_input=True, help="Password on pulp server")
 @click.option("--password", default="password", help="Password on pulp server")
-@click.option(
-    "--verify-ssl/--no-verify-ssl", default=True, help="Verify SSL connection"
-)
-@click.option(
-    "--format", type=click.Choice(["json"], case_sensitive=False), default="json"
-)
+@click.option("--verify-ssl/--no-verify-ssl", help="Verify SSL connection")
+@click.option("--format", type=click.Choice(["json"], case_sensitive=False))
 @click.pass_context
-def main(ctx, base_url, user, password, verify_ssl, format):
+def main(ctx, config, base_url, user, password, verify_ssl, format):
     ctx.ensure_object(PulpContext)
+
+    ctx.obj.config = PulpConfig(config)
     ctx.obj.api = OpenAPI(
-        base_url=base_url,
+        base_url=(base_url or ctx.obj.config.base_url or "https://localhost"),
         doc_path="/pulp/api/v3/docs/api.json",
         username=user,
```

Comment:
The name should also be part of the config, wdyt?
from name and host, netrc can provide the password

Reply:
I couldn't figure out how to get this to work. I tried to set auth on the requests session a few different ways (`self._session.auth = (username, None)`, `self._session.auth = (username, '')`, etc) but it seems that the only way to have it use `.netrc` is not to set `_auth` at all so I am not using defaults for username and password and that seems to work.